### PR TITLE
Audit: escape the formula name when checking the desc

### PR DESCRIPTION
**Pull request by bfontaine**

### CODE REVIEW

File: Library/Homebrew/cmd/audit.rb [hunk 1]
```diff
@@ -411,7 +411,7 @@ def audit_desc
       problem "Description shouldn't start with an indefinite article (#{$1})"
     end
 
-    if desc =~ /^#{formula.name}\s/i
+    if desc =~ /^#{Regexp.escape formula.name}\s/i
```
Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: Can we use `start_with?` directly?  e.g. `desc.downcase.start_with? "#{formula.name} "`

The main reason is regex has poor performance compared with direct string matching.

Comment (bfontaine) [inline Library/Homebrew/cmd/audit.rb]: Good idea, I just modified the PR to use `start_with?`.

Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: It looks like that many existed audits can be rewrite to use `start_with?` or `include?`.

Comment (bfontaine) [inline Library/Homebrew/cmd/audit.rb]: Yeah, I’ll merge that one quickly to fix #48744 and will look at the others just after.

Comment (xu-cheng) [inline Library/Homebrew/cmd/audit.rb]: :bow: :+1:

**Comment by bfontaine** [comment]
@BrewTestBot test this please.

**Pull request closed without merging**